input/elasticsearch.js

```diff
@@ -0,0 +1,59 @@
+const elasticsearch = require('elasticsearch')
+
+function create ({config, query, parser}) {
+  return {
+    name: 'Elasticsearch',
+    start: (pipeline) => {
+      const client = new elasticsearch.Client(config)
+      // Keep track of the processed ids
+      let ids = []
+      // Only run once
+      let running = false
+      // Run if not running
+      const run = () => {
+        if (!running) {
+          running = true
+          poll()
+        }
+      }
+      // Done
+      const done = () => {
+        running = false
+        setInterval(run, 333)
```

Comment:
It looks like you're scheduling an interval every time you run. Which explain why you need to check if something is already running. But I'm afraid it might lead to a memory leak. We're never clearing all those intervals.

Why don't we call `setInterval` only once or call `setTimeout` after each run?

Reply:
Sorry, that's supposed to be a setTimeout!

The intent is 1 concurrent request, 1/3 second between requests.

Reply:
Cool.